src/v2/models/AppState.js

```diff
@@ -77,7 +77,7 @@ export default Model.extend({
         Logger.warn(`Cannot find view state for form ${currentFormName}. Fall back to terminal state.`);
       }
 
-      currentViewState = this.get('terminal');
+      currentViewState = this.get('currentState').success ? this.get('currentState').success : this.get('terminal');
```

Comment:
- does `success` still exist as first citizen when `success` is a form in API?
- instead of changing formName in another function, can we do here like
```
if step === success then return 
{
  name: 'success-redirect',
}
```

Reply:
I see. the `success` is there by accident (or issue in the transformation layer). probably shall't rely on it.

Reply:
success does not exist as a firstLevel citizen when we are trying to set `currentViewState `. Thats why i had to reach into the currentState object.

We are not really changing the formName right, we are just reading it from `resp.currentState.success.name` since there is not remediation for the success form. We can avoid checking for the step here then.

Reply:
then do you need the changes at line 101 below?

Reply:
yes, thats where we set `currentFormName`

Reply:
`currentForm` is used in `getCurrentViewState` to get form data from remediation. since you dont use `formName` at all during the `success`, I doubt we need to change anythings in `setIonResponse`?

Reply:
I think we can fix the responseTransformer so that we don't have to read form name from resp.currentState.success.name